modules/cnf-programming-the-n3000-operator.adoc

```diff
@@ -0,0 +1,339 @@
+    - MAC: 64:4c:36:11:1b:ac
+      NVMVersion: 7.00 0x800052b0 0.0.0
+      PCIAddr: 0000:1c:00.0
+      name: Ethernet Controller XXV710 Intel(R) FPGA Programmable Acceleration Card N3000 for Networking
+    - MAC: 64:4c:36:11:1b:ad
+      NVMVersion: 7.00 0x800052b0 0.0.0
+      PCIAddr: 0000:1c:00.1
+      name: Ethernet Controller XXV710 Intel(R) FPGA Programmable Acceleration Card N3000 for Networking
+  fpga:
+  - PCIAddr: 0000:1b:00.0 <1>
+    bitstreamId: "0x23000410010310" <2>
+    bitstreamVersion: 0.2.3
+    deviceId: "0x0b30"
+----
+<1> The `PCIAddr` field indicates the PCI address of the card.
+<2> The `bitstreamId` field indicates the bitstream that is currently stored in flash.
+
+. Save the current `bitstreamId`, `PCIAddr`, the name, and the `deviceId` without "0x" padding.
++
+[source,terminal]
+----
+$ oc get n3000node -o json
+----
+
+. Update the user bitstream of the Intel FPGA PAC N3000 card:
+
+.. Define the N3000 cluster resource to program by creating a file named `n3000-cluster.yaml` as shown in the following example:
++
+[source,yaml]
+----
+apiVersion: fpga.intel.com/v1
+kind: N3000Cluster
+metadata:
+    name: n3000 <1>
+    namespace: vran-acceleration-operators
+spec:
+    nodes:
+      - nodeName: "node1" <2>
+        fpga:
+          - userImageURL: "http://10.10.10.122:8000/pkg/20ww27.5-2x2x25G-5GLDPC-v1.6.1-3.0.0_unsigned.bin" <3>
+            PCIAddr: "0000:1b:00.0" <4>
+            checksum: "0b0a87b974d35ea16023ceb57f7d5d9c" <5>
+----
+<1> Specify the name. The name must be `n3000`.
+<2> Specify the node to program.
+<3> Specify the URL for the user bitstream. This bitstream file must be accessible on an HTTP or HTTPS server.
+<4> Specify the PCI address of the card to program.
+<5> Specify the MD5 checksum of the bitstream that is specified in the `userImageURL` field.
++
+The N3000 daemon updates the FPGA user bitstream using the Open Programmable Acceleration Engine (OPAE) tools and resets the PCI device.
+The update of the FPGA user bitstream can require up to 40 minutes per card. For programming cards on multiple nodes, the programming happens one node at a time.
+
+.. Apply the update to begin programming the card with the bitstream:
++
+[source,terminal]
+----
+$ oc apply -f n3000-cluster.yaml
+----
++
+The N3000 daemon starts programming the bitstream after the appropriate 5G FEC user bitstream has been provisioned, such as `20ww27.5-2x2x25G-5GLDPC-v1.6.1-3.0.0_unsigned.bin` in this example, and after the CR has been created.
+
+.. Check the status:
++
+[source,terminal]
+----
+oc get n3000node
+----
++
+.Example output
+[source,terminal]
+----
+NAME             FLASH
+node1            InProgress
+----
+
+. Check the logs:
+.. Determine the pod name of the N3000 daemon:
++
+[source,terminal]
+----
+$ oc get pod -o wide | grep n3000-daemonset | grep node1
+----
++
+.Example output
+[source,terminal]
+----
+n3000-daemonset-5k55l              1/1     Running   0          15d
+----
+.. View the logs:
```

Comment:
Please add a line above this step to match the convention with other files in the repository.